[TRTC-102][docs] `--extra_llm_api_options`->`--config` in docs/examples/tests

venkywonka opened the pull request:
## Description

- [x] Update user-facing docs and examples to use `--config` instead of `--extra_llm_api_options`. Keep `--extra_llm_api_options` documented once as a backwards-compatible alias.
- [x] Make all the integration tests use `--config` (although they aren't user-facing)
- [x] Fix a bug I noticed where due to lack of `__init__.py` files in examples and database subdirs, their imports were failing when running `test_generate_config.py`.
- [x] Introduce a shared note that's included in all the high-traffic reference pages (trtllm-serve, trtllm-eval, trtllm-bench, features/*.md docs...) like below - to make it clear to the users that the `--config` is preferred but is non-breaking:
<img width="2204" height="588" alt="CleanShot 2025-12-16 at 17 53 29@2x" src="https://github.com/user-attachments/assets/614b3229-0a0f-491b-becb-0ef65159fa5d" />

- [x] The current docs build workflow auto-replaces all `x.y.z` with the current TRTLLM versions. This shouldn't happen when unintended, and devs must have a way to opt-out. So introduced a `TRTLLM_DOCS_REPLACE_CONTAINER_TAG` env var that can be opted-out (defaults to 1 so that it doesn't break existing doc building pipelines)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Updated configuration mechanism documentation to reflect renamed CLI option from `--extra_llm_api_options` to `--config` across deployment guides, examples, and benchmark documentation.
  * Updated associated configuration file naming conventions for consistency.
  * Added backward compatibility note indicating the previous flag name remains supported as an alias.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

<!--
Please explain the issue and the solution in short.
-->

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

venkywonka committed:
[None][docs] Prefer --config in docs/examples

Update user-facing docs and examples to use --config instead of --extra_llm_api_options.
Keep --extra_llm_api_options documented once as a backwards-compatible alias.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Use consistent config.yml filenames in docs and examples

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
Use --config flag in integration tests for coverage

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run --disable-fail-fast

anish-shanbhag reviewed (approved):
Thanks for the UX improvement 👍 

Should we also update this line? https://github.com/venkywonka/TensorRT-LLM/blob/flag-docs-update-fTGJG/scripts/generate_config_table.py#L136

tensorrt-cicd commented:
[PR_Github #28440](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/28440/) [ run ] completed with state `FAILURE`. Commit: `8798b78` 
[/LLM/main/L0_MergeRequest_PR pipeline #21769](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/21769/) completed with status: 'FAILURE'

yechank-nvidia reviewed (approved):
LTGM.

venkywonka committed:
docs: centralize YAML config note and prefer --config

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
fix: add __init__.py files and fix imports for test_generate_config_table

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented on docs/source/commands/trtllm-serve/run-benchmark-with-trtllm-serve.md:
````diff
@@ -44,7 +50,7 @@ TensorRT LLM distributes the pre-built container on [NGC Catalog](https://catalo
 
 ```bash
-docker run --rm -it --ipc host -p 8000:8000 --gpus all --ulimit memlock=-1 --ulimit stack=67108864 nvcr.io/nvidia/tensorrt-llm/release:x.y.z
+docker run --rm -it --ipc host -p 8000:8000 --gpus all --ulimit memlock=-1 --ulimit stack=67108864 nvcr.io/nvidia/tensorrt-llm/release:1.2.0rc6
````
x.y.z was replaced by 1.2.0rc6 inadvertently by the docs building workflow. will have to revert this.

venkywonka committed:
docs: keep x.y.z container tag placeholder

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
docs: improve run-benchmark-with-trtllm-serve TOC and --config section

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
docs: add --config alias note to feature and benchmarking docs

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run --disable-multi-gpu-test

tensorrt-cicd commented:
[PR_Github #28820](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/28820/) [ run ] completed with state `SUCCESS`. Commit: `3da9be9` 
[/LLM/main/L0_MergeRequest_PR pipeline #22064](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/22064/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #28945](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/28945/) [ run ] triggered by Bot. Commit: `14e8cc3`

venkywonka committed:
docs: default TRTLLM_DOCS_REPLACE_CONTAINER_TAG to 1

This keeps CI/pipeline docs builds replacing the `nvcr.io/nvidia/tensorrt-llm/release:x.y.z`
placeholder unless explicitly opted out.

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
reconcile with conflicts with #9832

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

venkywonka committed:
chore: fix rebase conflict, deslop

Signed-off-by: Venky Ganesh <23023424+venkywonka@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #29137](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29137/) [ run ] triggered by Bot. Commit: `d365dda`

tensorrt-cicd commented:
[PR_Github #29136](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29136/) [ run ] completed with state `ABORTED`. Commit: `27c842c`

tensorrt-cicd commented:
[PR_Github #29137](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29137/) [ run ] completed with state `SUCCESS`. Commit: `d365dda` 
[/LLM/main/L0_MergeRequest_PR pipeline #22344 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/22344/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

tensorrt-cicd commented:
[PR_Github #29148](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29148/) [ run ] triggered by Bot. Commit: `d365dda`

tensorrt-cicd commented:
[PR_Github #29172](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29172/) [ skip ] triggered by Bot. Commit: `d365dda`

tensorrt-cicd commented:
[PR_Github #29172](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/29172/) [ skip ] completed with state `SUCCESS`. Commit: `d365dda` 
Skipping testing for commit `d365dda`
